improver/utilities/spatial.py

```diff
@@ -214,44 +233,93 @@ def create_difference_cube(self, cube, coord_name, diff_along_axis):
         """
         Calculate the difference along the axis specified by the
         coordinate.
 
         Args:
             cube (iris.cube.Cube):
                 Cube from which the differences will be calculated.
-            coord_axis (str):
-                Short-hand reference for the x or y coordinate, as allowed by
-                iris.util.guess_coord_axis.
+            coord_name (str):
+                Name of coordinate along which the difference is calculated.
 
         Returns:
-            iris.cube.Cube:
-                Cube after the differences have been calculated along the
+           np.ndarray:
+                Array after the differences have been calculated along the
                 specified axis.
         """
-        coord_name = cube.coord(axis=coord_axis).name()
         diff_axis = cube.coord_dims(coord_name)[0]
         diff_along_axis = np.diff(cube.data, axis=diff_axis)
-        diff_cube = self.create_difference_cube(cube, coord_name, diff_along_axis)
-        return diff_cube
+        return diff_along_axis
+
+    def process(self, cube):
+        """
+        Calculate the difference along the x and y axes and return
+        the result in separate cubes. The difference along each axis is
+        calculated using numpy.diff.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube from which the differences will be calculated.
+
+        Returns:
+            (tuple): tuple containing:
+                **diff_along_y_cube** (iris.cube.Cube):
+                    Cube after the differences have been calculated along the
+                    y axis.
+                **diff_along_x_cube** (iris.cube.Cube):
+                    Cube after the differences have been calculated along the
+                    x axis.
 
-    @staticmethod
-    def gradient_from_diff(diff_cube, ref_cube, coord_axis):
+        """
+        diffs = []
+        for axis in ["x", "y"]:
+            coord_name = cube.coord(axis=axis).name()
+            diff_cube = self.create_difference_cube(
+                cube, coord_name, self.calculate_difference(cube, coord_name)
+            )
+            self._update_metadata(diff_cube, coord_name, cube.name())
+            diffs.append(diff_cube)
+        return diffs[0], diffs[1]
+
+
+class GradientBetweenAdjacentGridSquares(BasePlugin):
+
+    """Calculate the gradient between adjacent grid squares within
+    a cube. The gradient is calculated along the x and y axis
+    individually."""
+
+    def __init__(self, regrid=False):
+        """Initialise plugin.
+
+        Args:
+            regrid (bool):
+                If True, the gradient cube is regridded to match the spatial
+                dimensions of the input cube. If False, the length of the
+                spatial dimensions of the gradient cube are one less than for
+                the input cube.
+            """
+        self.regrid = regrid
+
+    def _update_metadata(self, gradient, cube):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5583761)